src/edge_cov.rs

```diff
@@ -0,0 +1,107 @@
+use alloy_primitives::{Address, U256};
+use revm::{
+    interpreter::{opcode::OpCode, Interpreter},
+    Database, EvmContext, Inspector,
+};
+
+use std::hash::{Hash, Hasher};
+
+/// A hit count that never goes to zero e.g. the back edge of a loop that iterates 256 times will be
+/// one instead of zero.
+// see https://github.com/AFLplusplus/AFLplusplus/blob/5777ceaf23f48ae4ceae60e4f3a79263802633c6/instrumentation/afl-llvm-pass.so.cc#L810-L829
+#[derive(Clone, Copy, Debug, Default)]
+pub struct NeverZeroHitCount(pub u8);
+
+impl std::ops::Add for NeverZeroHitCount {
+    type Output = Self;
+
+    fn add(self, rhs: Self) -> Self {
+        Self(self.0.checked_add(rhs.0).unwrap_or(1))
+    }
+}
+
+impl std::ops::AddAssign for NeverZeroHitCount {
+    fn add_assign(&mut self, rhs: Self) {
+        *self = *self + rhs;
+    }
+}
+
+const MAX_EDGE_COUNT: usize = 65536;
+
+/// An Inspector that tracks edge coverage.
```

Comment:
not immediately obvious what this means.
can we add a few more docs here that describe what this records?